### PR TITLE
Fix remaining issues in Dotty codebase

### PR DESCRIPTION
**Pull request by tgodzik**
- fix issue with operators with literal on the next line
- fix parsing of extension methods with a comment and no `:`

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -295,7 +295,9 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
           if (indentedRegion(sepRegionsParameter) && currentIndent >= 0) {
             var sepRegionsProcess = sepRegionsParameter
             while ((indentedRegion(sepRegionsProcess)
-                && sepRegionsProcess.head.indent > currentIndent) ||
+                && sepRegionsProcess.head.indent > currentIndent && !isLeadingInfixOperator(
```
Comment (kpbochenek) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:298]: good catch! :+1:

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 2]
```diff
@@ -3710,7 +3713,8 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     } else if (token.is[LeftBrace]) {
       inBraces(templateStats())
     } else {
-      newLineOpt()
+      while (token.is[LF] || token.is[LFLF])
```
Comment (kpbochenek) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:3716]: tokenizer should never produce 2 consecutive LF / LFLF combinations:
2x LF => LFLF
2x LFLF => LFLF

`while` should not be needed, `newLinesOpt()` should be enough

**Review by kpbochenek — APPROVED** [review]
Nice work fixing those issues! :+1:

(one minor cosmetic sugestion - can be ignored :) )

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 2 above)
Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala]: Changed and added more newlines, thanks!

**Pull request merged**